react/components/ProductImages/components/Carousel/ThumbnailSwiper.js

```diff
@@ -56,63 +41,146 @@ const Thumbnail = ({
       <div
         className={`absolute absolute--fill b--solid b--muted-2 bw0 ${handles.carouselThumbBorder}`}
       />
-    </div>
+    </>
   )
 }
 
-const ThumbnailSwiper = ({
-  isThumbsVertical,
-  slides,
-  swiperParams,
-  thumbUrls,
-  position,
-  onThumbClick,
-  activeIndex,
-  thumbnailAspectRatio,
-  thumbnailMaxHeight,
-}) => {
-  const hasThumbs = slides.length > 1
+const navigationConfig = {
+  prevEl: '.swiper-thumbnails-caret-prev',
+  nextEl: '.swiper-thumbnails-caret-next',
+  disabledClass: `c-disabled o-0 pointer-events-none ${styles.carouselCursorDefault}`,
+  hiddenClass: 'dn',
+}
+
+const ThumbnailSwiper = props => {
   const handles = useCssHandles(CSS_HANDLES)
 
-  const thumbClasses = classNames(`${handles.carouselGaleryThumbs} dn h-auto`, {
-    'db-ns': hasThumbs,
-    mt3: !isThumbsVertical,
-    'w-20 bottom-0 top-0 absolute': isThumbsVertical,
-    'left-0': isThumbsVertical && position === THUMBS_POSITION_HORIZONTAL.LEFT,
-    'right-0':
-      isThumbsVertical && position === THUMBS_POSITION_HORIZONTAL.RIGHT,
-  })
+  const {
+    isThumbsVertical,
+    slides,
+    thumbUrls,
+    position,
+    thumbnailAspectRatio,
+    thumbnailMaxHeight,
+    displayThumbnailsArrows,
+    ...swiperProps
+  } = props
+
+  const hasThumbs = slides.length > 1
+
+  const thumbClassName = classNames(
+    `${handles.carouselGaleryThumbs} dn h-auto`,
+    {
+      'db-ns': hasThumbs,
+      mt3: !isThumbsVertical,
+      'w-20 bottom-0 top-0 absolute': isThumbsVertical,
+      'left-0':
+        isThumbsVertical && position === THUMBS_POSITION_HORIZONTAL.LEFT,
+      'right-0':
+        isThumbsVertical && position === THUMBS_POSITION_HORIZONTAL.RIGHT,
+    }
+  )
+
+  const itemContainerClassName = classNames(
+    handles.productImagesThumb,
+    'mb5 pointer',
+    {
+      'w-20': !isThumbsVertical,
+      'w-100': isThumbsVertical,
+    }
+  )
+
+  const arrows = useMemo(() => {
+    if (!displayThumbnailsArrows) {
+      return null
+    }
+
+    const thumbCaretSize = 24
+    const thumbCaretClassName = `${handles.productImagesThumbCaret} absolute z-2 pointer c-action-primary flex pv2`
+    const thumbCaretStyle = { transition: 'opacity 0.2s' }
+
+    const nextBtnClassName = classNames(
+      'swiper-thumbnails-caret-next',
+      thumbCaretClassName,
+      {
+        [`bottom-0 pt7 left-0 justify-center w-100 ${styles.gradientBaseBottom}`]: isThumbsVertical,
+        [`right-0 top-0 items-center h-100 pl6 ${styles.gradientBaseRight}`]: !isThumbsVertical,
+      }
+    )
+
+    const prevBtnClassName = classNames(
+      'swiper-thumbnails-caret-prev top-0 left-0',
+      thumbCaretClassName,
+      isThumbsVertical && `pb7 justify-center w-100 ${styles.gradientBaseTop}`,
+      !isThumbsVertical && `items-center h-100 pr6 ${styles.gradientBaseLeft}`
```

Comment:
How about

```typescript
isThumbsVertical ?
  `pb7 justify-center w-100 ${styles.gradientBaseTop}` 
    : `items-center h-100 pr6 ${styles.gradientBaseLeft}`,
```

?

Reply:
Your heart is in the right place, but that extra indentation gives me the creeps 😳

Reply:
You could use the object of `classnames` instead
```suggestion
      {
        [`pb7 justify-center w-100 ${styles.gradientBaseTop}`]: isThumbsVertical,
        [`items-center h-100 pr6 ${styles.gradientBaseLeft}`]: !isThumbsVertical,
      }
```

Reply:
Why would that be better 😭 ?